httpserver/service_node.cpp

```diff
@@ -229,7 +258,14 @@ bool ServiceNode::process_store(const message_t& msg) {
     return true;
 }
 
-void ServiceNode::process_push(const message_t& msg) { save_if_new(msg); }
+void ServiceNode::process_push(const message_t& msg) {
+#ifndef DISABLE_POW
+    char* error_msg;
+    if (!verify_message(msg, error_msg))
+        throw std::runtime_error(error_msg);
```

Comment:
Seems a bit too harsh. Just ignore the message, provided that we printed appropriate warnings?

Reply:
As said above, this way the error message bubbles up and is returned in the response body.

Reply:
OK